### DISCUSSION TITLE
try same data in two locations in serenity

### DISCUSSION THREAD

**Discussion by syedbasit24** [serenity-is/Serenity#6476] [Help]
Suppose added some record in the Designation screen for PHPL(HQ) location, and for PHSB location in the Designation screen if try to add record with same code/Description which is already used in the PHLP location then it's not allowed to use same code/Description.

**Comment by edwardch** [comment]
If the records are in the same table you can add a unique index to the table.

If the records are in two different tables then you can override the `ValidateRequest()` in your SaveHandler.

Add something like this:

```cs
var hasCode = UnitOfWork.Connection.Exists<MyRow>(
    new Criteria(flds.Code) == Row.Code.Value
);

if (hasCode)
	throw new ValidationError($"{Row.Code.Value} already exists");
```

**Reply by syedbasit24** [reply]
Table is same but it must be show according to foreign key ( id=1 for one location and id= 2 for other locations)
We have to allow inserting same data in same screen but different locations

**Reply by edwardch** [reply]
Then add a unique index on both columns:

```sql
ALTER TABLE [Locations] ADD CONSTRAINT [UQ_Locations_LocationId_Code] UNIQUE NONCLUSTERED 
(
	[LocationId] ASC,
	[Code] ASC
)
```

**Reply by syedbasit24** [reply]
there is already a foreign key created  and primary key , verify only in code

**Reply by edwardch** [reply]
If you provide a bit more information we can try to assist

**Reply by stixoffire** [reply]
I am amazed edward was able to provide that much as there is no code describing the scenario of what you have and what you are trying to achieve. (I mean with specifics not generalizations)

Both fields are used as primary key or ?? 
If both fields are there should be a unique constraint ..
How is the table ?
LocationID (1 or 2 )
CodeName  (PL or SB)
Description  cannot be duplicated between locations but in same location it can be ??? 

Please describe your issue more clearly with table definition of some sort and then the rules ..(requirements)
